scripts/calculateRevenue/protocols/rhino/index.ts

```diff
@@ -106,29 +107,40 @@ export async function getTotalRevenueUsdFromBridges({
     const tokenDecimals = tokenContract
       ? BigInt(await tokenContract.read.decimals())
       : NATIVE_TOKEN_DECIMALS
-
+    let tokenPriceUsd = 0
     try {
       // Get the historical token prices
       const tokenPrices = await fetchTokenPrices({
         tokenId,
         startTimestamp,
         endTimestampExclusive: endTimestampExclusive,
       })
-      const tokenPriceUsd = getTokenPrice(tokenPrices, bridge.timestamp)
-      const partialUsdContribution =
-        Number(
-          (bridge.amount *
-            BigInt(tokenPriceUsd * 10 ** BRIDGE_VOLUME_USD_PRECISION)) /
-            10n ** tokenDecimals,
-        ) /
-        10 ** BRIDGE_VOLUME_USD_PRECISION
-      totalUsdContribution += partialUsdContribution
+      tokenPriceUsd = getTokenPrice(tokenPrices, bridge.timestamp)
     } catch (error) {
-      console.error(
-        `Error fetching token prices for ${tokenId} at ${bridge.timestamp}:`,
-        error,
-      )
+      try {
+        // If failed to get price from blockchain-api (probably because token is not supported)
+        // Try to get the historical price directly from coingecko
+        // Doing this second because it is less accurate (only daily snapshot at midnight UTC)
+        tokenPriceUsd = await getTokenHistoricalPrice({
```

Comment:
should we just use this?  it's seems hard to describe how we might calculate the KPI if we're like "well, we use this data if it's on this list over here that might change, and otherwise we use this data".